src/main/java/seedu/address/logic/parser/DeleteReminderCommandParser.java

```diff
@@ -0,0 +1,58 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.*;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.AddReminderCommand;
+import seedu.address.logic.commands.DeleteReminderCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.date.ReminderDate;
+import seedu.address.model.reminder.ReminderDescription;
+import seedu.address.model.reminder.ReminderList;
+
+import java.util.stream.Stream;
+
+/**
+ * Parses input arguments and creates a new DeleteReminderCommand object.
+ */
+public class DeleteReminderCommandParser implements Parser<DeleteReminderCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the DeleteReminderCommand
+     * and returns a DeleteReminderCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public DeleteReminderCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args,
+                        PREFIX_REMINDER);
+
+        Index index;
+        ReminderDescription reminderDescription;
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_REMINDER)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteReminderCommand.MESSAGE_USAGE));
+        }
+
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+            reminderDescription = ParserUtil.parseReminderDescription(argMultimap.getValue(PREFIX_REMINDER).get());
+        } catch (seedu.address.logic.parser.exceptions.ParseException pe) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddReminderCommand.MESSAGE_USAGE), pe);
+        }
+
+        return new DeleteReminderCommand(index, reminderDescription);
+    }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
```

Comment:
I'm guessing this is because you're trying to support multiple deletes in one command? Need to test if using other prefixes will mess up the list

Reply:
Add tests